### PR TITLE
[llama3] Enable a llama3 model.

### PR DESCRIPTION
**Pull request by stellaraccident**
* I had a NYI assert for GQA. Fixes that.
* Just grabs some pre-made weights of unknown providence to test for gross correctness.

### CODE REVIEW

**Comment by aviator19941** [comment]
@stellaraccident When I run `python -m sharktank.examples.export_paged_llm_v1 --hf-dataset=llama3_8B_q8_0`, it yields this [error](https://gist.github.com/aviator19941/abb9d210aebe29307d2ff6f863878a19). Is there another way you ran this?

File: sharktank/sharktank/types/theta.py [hunk 1]
```diff
@@ -537,11 +537,11 @@ def _dataset_load_helper(
 ) -> Dataset:
     path = Path(path)
     suffixes = path.suffixes
-    if file_type == "gguf" or suffixes == [".gguf"]:
+    if file_type == "gguf" or suffixes[-1] == ".gguf":
         from . import gguf_interop
 
         return gguf_interop.load_file(path)
-    elif file_type == "irpa" or suffixes == [".irpa"]:
+    elif file_type == "irpa" or suffixes[-1] == ".irpa":
```
Comment (ScottTodd) [inline sharktank/sharktank/types/theta.py:544]: Oh, I ran into this suffix issue on my own and sent a PR: https://github.com/nod-ai/sharktank/pull/15

Can we get some of these merged?

Comment (stellaraccident) [reply sharktank/sharktank/types/theta.py:544]: Please do. PRs are piling up. It's there an access issue we need to fix?

File: sharktank/sharktank/models/llama/llama.py [hunk 2]
```diff
@@ -305,10 +305,6 @@ def forward(
                 xq=xq, xk=xk, mask=embedding_batch_mask
             )
 
-        # TODO: Some model variants do some form of kv repetition to expand the
-        # count of kv heads to the count of attention heads used by the q.
-        assert self.head_count == self.head_count_kv, "NYI: KV expansion"
```
Comment (ScottTodd) [inline sharktank/sharktank/models/llama/llama.py:310]: Thanks, this also helped me get `mistral-7b-v0.1.Q8_0.gguf` from https://huggingface.co/TheBloke/Mistral-7B-v0.1-GGUF running!

**Review by ScottTodd — APPROVED** [review]

File: sharktank/sharktank/models/llama/llama.py (hunk 2 above)
Comment (stellaraccident) [reply sharktank/sharktank/models/llama/llama.py:310]: Take over/land it off you have a test setup that works.

Comment (ScottTodd) [reply sharktank/sharktank/models/llama/llama.py:310]: Tested `llama3_8B_q8_0` and got it to complete `python -m sharktank.examples.paged_llm_v1 --hf-dataset=llama3_8B_q8_0 "1, 2, 3, 4"` with output like
```
:: Result tokens: [[11, 220, 20, 11, 220, 21, 11, 220, 22]]
  0(9, 19): , 5, 6, 7
```
Good enough? Not sure how else we want to validate.

Comment (ScottTodd) [reply sharktank/sharktank/models/llama/llama.py:310]: Well... dunno what to expect, some of these aren't commas :P

```
:: Result tokens: [[11, 220, 20, 11, 220, 21, 11, 220, 22, 11, 220, 23, 0, 220, 24, 11, 220, 605, 11, 220, 806, 11, 220, 717, 11, 220, 1032, 0, 220, 975, 11]]
  0(31, 41): , 5, 6, 7, 8! 9, 10, 11, 12, 13! 14,
```

**Comment by ScottTodd** [comment]
CI failure looks unrelated. Wonder if the serving tests are flaky
https://github.com/nod-ai/sharktank/actions/runs/8943283629/job/24567652312?pr=4#step:7:512

File: sharktank/sharktank/models/llama/llama.py (hunk 2 above)
Comment (stellaraccident) [reply sharktank/sharktank/models/llama/llama.py:341]: Good enough for now. We need to do a later by layer comparison and test. Did that a month ago and needs someone to go through the details again. Part of the bringup process.

**Comment by stellaraccident** [comment]
> CI failure looks unrelated. Wonder if the serving tests are flaky https://github.com/nod-ai/sharktank/actions/runs/8943283629/job/24567652312?pr=4#step:7:512

Yeah, flaky

**Comment by ScottTodd** [comment]
> > CI failure looks unrelated. Wonder if the serving tests are flaky https://github.com/nod-ai/sharktank/actions/runs/8943283629/job/24567652312?pr=4#step:7:512
> 
> Yeah, flaky

Sent https://github.com/nod-ai/sharktank/pull/17 to hopefully fix the flake.

**Pull request merged**